modules/connatixBidAdapter.js

```diff
@@ -140,10 +155,12 @@ export const spec = {
 
     _handleEids(requestPayload, validBidRequests);
 
+    context = requestPayload;
```

Comment:
So you are wanting to send your bidRequest payload in every single event you fire?

Seems like a lot of duplication but it is fine of course.

The only thing I can think of is that there can be race condition here if the publisher site is running multiple auctions within close time period.

Auction A
Auction B

Context = {};
Auction A start
Context = A

Auction B Start
Context = B

Auction A Bid WON / timeout etc
=> Event sends Context = Auction B for an event from Auction A

We can get around this by perhaps making context a map of Auction IDs => context

Then sending the right one in each event.

Or if this is enough of an edge case and you would rather keep it more simple then that is fine.

Just a suggestion!

Reply:
Thanks for the thoughtful suggestion! We did discuss this model with the backend team, and they specifically need the bidRequest payload with each event for their current setup.

We understand the potential for race conditions and have considered various options. However, for now, we’ll stick with this approach to meet the backend requirements.

I appreciate your input and am grateful for your understanding. If you have any more thoughts or questions, feel free to share!